cheatsheets/gleam-for-go-users.md

```diff
@@ -0,0 +1,1452 @@
+
+Comments starting with `///` are used to document the following function,
+constant, or type definition. Comments starting with `////` are used to
+document the current module.
+
+```gleam
+//// This module is very important.
+
+/// The answer to life, the universe, and everything.
+const answer: Int = 42
+
+/// A main function
+fn main() {}
+
+/// A Dog type
+type Dog {
+  Dog(name: String, cuteness: Int)
+}
+```
+
+`//` comments are not used while generating documentation files, while
+`////` and `///` will appear in them.
+
+## Variables
+
+You can rebind variables in both languages.
+
+### Go
+
+In Go you specify values by:
+- Specifically declare the variable with the `var` keyboard and a name and then either just the type, or an optional type and value
+- a variable name (or comma separated list of names), the `:=` operator and then an expression
+
+```Go
+size := 50
+size := size + 100
+size = 1
+```
+
+### Gleam
+
+Gleam has the `let` keyword before its variable names.
+
+```gleam
+let size = 50
+let size = size + 100
+let size = 1
+```
+
+### Match operator
+
+#### Go
+
+Go has no real support for any sort of pattern matching or destructuring.  It does support multiple return values.  These are assigned to
+separated variable names separated by commas
+
+```Go
+a, b := func() (int, int) { return 1, 2 }();
+// a == 1
+// b == 2
+```
+
+#### Gleam
+
+In Gleam, `let` and `=` can be used for pattern matching, but you'll get
+compile errors if there's a type mismatch, and a runtime error if there's
+a value mismatch. For assertions, the equivalent `let assert` keyword is
+preferred.
+
+```gleam
+let #(a, _) = #(1, 2)
+// a = 1
+// `_` matches 2 and is discarded
+
+let assert [] = [1] // runtime error
+let assert [y] = "Hello" // compile error, type mismatch
+```
+
+Asserts should be used with caution.
+
+### Variables type annotations
+
+#### Go
+
+In Go type annotations are generally required in most places.  The exception is when variables or constants are initialized with a value. Multiple consecutive struct fields or function arguments of the same time can combined their type annotation. 
+
+```Go
+type SomethingElse struct {
+  field string
+  otherField int
+  x, y float64
+}
+
+func multipleArgsOfSameType(str string, x, y float64) {}
+```
+
+#### Gleam
+
+In Gleam type annotations can optionally be given when binding variables.
+
+```gleam
+let some_list: List(Int) = [1, 2, 3]
+let some_string: String = "A string"
+```
+
+Gleam will check the type annotation to ensure that it matches the type of the
+assigned value. It does not need annotations to type check your code, but you
+may find them useful for documentation purposes.
+
+## Functions
+
+### Go
+
+In Go, you can define functions with the `func` keyword. The `return`
+keyword is required if the function has a return value, and optional otherwise.
+
+```Go
+func hello(name string) string {
+  if name == 'Joe' {
+    return 'Welcome back, Joe!'
+  }
+  return "Hello $name"
+}
```

Comment:
You need double quotes for strings. Single quotes are runes (characters). Go doesn't have interpolation inside of strings. 

```suggestion
func hello(name string) string {
	if name == "Joe" {
		return "Welcome back, Joe!"
	}
	return fmt.Sprintf("Hello %s", name)
}
```

Reply:
Done